pkg/runtime/optimize/optimizer.go

```diff
@@ -242,11 +242,55 @@ func (o optimizer) optimizeShowDatabases(ctx context.Context, stmt *rast.ShowDat
 	return ret, nil
 }
 
+func (o optimizer) overwriteLimit(stmt *rast.SelectStatement, args []interface{}) (originLimit, overwriteLimit *rast.LimitNode) {
+	if stmt == nil || stmt.Limit == nil {
+		return nil, nil
+	}
+
+	offset := stmt.Limit.Offset()
+	limit := stmt.Limit.Limit()
+
+	originLimit = new(rast.LimitNode)
+	overwriteLimit = new(rast.LimitNode)
+
+	// SELECT * FROM student where uid = ? limit ? offset ?
+	if stmt.Limit.IsLimitVar() {
```

Comment:
There's another case, offset is variable, but limit is not. For example, `select * from student where ... limit ?, 10`.